frontend/public/components/routes/create-route.tsx

```diff
@@ -0,0 +1,447 @@
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formData: {
+        name: '',
+        hostname: '',
+        path: '',
+        service: '',
+        targetPort: '',
+        terminationType: '',
+        insecureTrafficType: '',
+        certificateFile: '',
+        privateKeyFile: '',
+        caCertificateFile: '',
+        destinationCaCertificateFile: '',
+        error: '',
+      },
+      toggleData: {
+        secure: false,
+      },
+      loaded: false,
+      inProgress: false,
+      namespace: getActiveNamespace(),
+      services: [],
+      labels: {},
+      portOptions: {},
+    };
+  }
+
+  componentDidMount() {
+    k8sList(ServiceModel, {ns: this.state.namespace})
+      .then(services => this.setState({
+        services,
+        loaded: true,
+      }))
+      .catch((err) => {
+        if (err.response.status !== 403) {
+          const errorMessage = err.message || 'Could not load services.';
+          this.setState({
+            formData: {
+              ...this.state.formData,
+              error: errorMessage
+            }
+          });
+        }
+      });
+  }
+
+  handleChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    this.setState({
+      formData: {
+        ...this.state.formData,
+        [event.currentTarget.name]: event.currentTarget.value
+      }
+    });
+  }
+
+  changeService = (serviceName: string) => {
+    const selectedService = _.find(this.state.services, { metadata: { name: serviceName } });
+    const portOptions = getPortOptions(selectedService);
+    const labels = selectedService.metadata.labels;
+    this.setState({
+      formData: {
+        ...this.state.formData,
+        service: serviceName,
+      },
+      labels,
+      portOptions,
+    });
+  }
+
+  changeTargetPort = (portName: string) => {
+    this.setState({
+      formData: {
+        ...this.state.formData,
+        targetPort: portName
+      }
+    });
+  }
+
+  toggleSection: React.ReactEventHandler<HTMLInputElement> = event => {
+    this.setState({
+      toggleData: {
+        ...this.state.toggleData,
+        [event.currentTarget.name]: event.currentTarget.checked
+      }
+    });
+  }
+
+  changeTerminationType = (terminationType: string) => {
+    this.setState({
+      formData: {
+        ...this.state.formData,
+        terminationType,
+        // unset insecureTrafficType if it was set
+        insecureTrafficType: '',
+      }
+    }, () => {
+      if (terminationType === 'passthrough') {
+        this.setState({
+          formData: {
+            ...this.state.formData,
+            // unset tls data if it was set
+            certificateFile: '',
+            privateKeyFile: '',
+            caCertificateFile: '',
+            destinationCaCertificateFile: '',
+          }
+        });
+      }
+    });
+  }
+
+  changeInsecureTrafficType = (insecureTrafficType: string) => {
+    this.setState({
+      formData: {
```

Comment:
I wonder if we should do away with `formData` and just make all these top-level properties. Then it can just be

```js
this.setState({ insecureTrafficType });
```